README.md

```diff
@@ -1,35 +1,43 @@
-### WARNING
-There will be some inevitable churn for the next week as I settle on the api!
+<div align="center">
 
-This is a WORK IN PROGRESS repo :)
+<h1>refactoring.nvim</h1>
+<h5>The Refactoring library based off the Refactoring book by Martin Fowler</h5>
 
-### Supported languages
+[![Lua](https://img.shields.io/badge/Lua-blue.svg?style=for-the-badge&logo=lua)](http://www.lua.org)
+[![Neovim](https://img.shields.io/badge/Neovim%200.6+-green.svg?style=for-the-badge&logo=neovim)](https://neovim.io)
 
-Look in [queries/](queries/) to see which languages are currently supported.
+</div>
 
-#### Things todo
-* I want to write tests for a bunch of stuff.
-* ...
-* stability and profitability
+## WIP
 
-# refactoring.nvim
-The Refactoring library based off the Refactoring book by Martin Fowler
+Although this plugin is usable, it is very much a work in progress, meaning that
+the API can change very rapidly.
 
-## Setup
+## Table of Contents
 
-**Requirements**:
+- [Installation](#installation)
+  - [Requirements](#requirements)
+  - [Packer](#packer)
+- [Features](#features)
+  - [Supported Languages](#supported-languages)
+  - [Refactoring Features](#refactoring-features)
+  - [Debug Features](#debug-features)
+- [Configuration](#configuration)
+  - [Configuration for Refactoring Operations](#config-refactoring)
+    - [Using Direct Remaps](#config-refactoring-direct)
+    - [Using Telescope](#config-refactoring-telescope)
+  - [Configuration for Debug Operations](#config-debug)
 
-**NEOVIM NIGHTLY**
-The plugin is tested on neovim nightly, please make sure you are on nightly
-before submitting an issue with the plugin.
+## Installation<a name="installation"></a>
 
-Plugins:
+### Requirements<a name="requirements"></a>
+
+- Neovim 0.6 Stable or 0.7 Nightly
```

Comment:
I think we should just say neovim nightly. Have already been bitten by some neovim upgrades with some issues being raised with pluign using new API for something.